content/blog/2022/2022-02-15-getting-started-with-spring-webflux.md

```diff
@@ -0,0 +1,566 @@
+
+{{% image alt="Non-blocking request" src="images/posts/spring-webflux/non-blocking_request.png" %}}
+
+### Reactive Libraries used in Java
+
+The landscape of reactive world in Java is quite evolving and matured enough. Every now and then we hear a new Reactive framework being introduced and consumed by different companies. Some of them include:
+
+* **RxJava**: It is implemented out of the [ReactorX](https://reactivex.io/) project which hosts implementations for multiple programming languages and platforms. *ReactiveX* is a combination of the best ideas from the *Observer* pattern, the *Iterator* pattern, and *functional programming*.
+* **Project Reactor**: [Reactor](https://projectreactor.io/) is a framework solely built by Pivotal and powered by Spring. It is considered as one of the foundation of the *reactive stack* in the [Spring](https://spring.io/reactive) ecosystem. It implements Reactive API patterns which is based out of *Reactive Streams* specification.
+* **Akka Streams**: Although it implements Reactive Streams implementation, the [Akka Streams](http://doc.akka.io/docs/akka/snapshot/scala/stream/index.html) API is completely decoupled from the Reactive Streams interfaces. It uses *Actors* to deal with the streaming data. It is considered as a 3rd generation Reactive library.
+* **Ratpack**: [Ratpack](https://ratpack.io/) is a set of Java libraries used for building scalable and high-performance HTTP applications. It uses Java 8, *Netty* and reactive principles to provide a basic implementation of Reactive Stream API. You can also use Reactor or RxJava along with it.
+* **Vert.x**: [Vert.x](https://vertx.io/) is a foundation project by *Eclipse* which delivers a *polyglot event driven framework* for JVM. It is similar to Ratpack and allows to use RxJava or their native implementation for Reactive Streams API.
+
+## Intro to Java 9 Reactive Streams API
+
+The whole purpose of Reactive Streams was to introduce a standard for asynchronous stream processing of data with non-blocking backpressure. Hence, few engineers from various companies like Netflix, Pivotal, Lightblend, RedHat, etc. joined together to implement the *Reactive Streams API*. It is implemented based upon the *Publisher-Subscriber* Model or *Producer-Consumer* Model and primarily defines four interfaces:
+
+* ***Publisher***: It is responsible for preparation and transfer of data to subscribers as individual messages. A Publisher can serve multiple subscribers but it has only one method, `subscribe`.
```

Comment:
when mentioning method names in text, please add parentheses to make it recognizable as a method. I.e. `subscribe()` instead of `subscribe`.

Reply:
Updated it as method.